src/error.rs

```diff
@@ -6,32 +6,41 @@ pub use self::audio_format::Error as AudioFormatError;
 pub use self::audio_unit::Error as AudioUnitError;
 use crate::OSStatus;
 
+use objc2_audio_toolbox::{
+    kAudioServicesSystemSoundClientTimedOutError, kAudioServicesSystemSoundUnspecifiedError,
+};
+
 pub mod audio {
     use crate::OSStatus;
+    use objc2_core_audio_types::{
+        kAudio_BadFilePathError, kAudio_FileNotFoundError, kAudio_FilePermissionError,
+        kAudio_MemFullError, kAudio_ParamError, kAudio_TooManyFilesOpenError,
+        kAudio_UnimplementedError,
+    };
 
     #[derive(Copy, Clone, Debug)]
     pub enum Error {
-        Unimplemented = -4,
-        FileNotFound = -43,
-        FilePermission = -54,
-        TooManyFilesOpen = -42,
-        BadFilePath = 561017960,
-        Param = -50,
-        MemFull = -108,
+        Unimplemented = kAudio_UnimplementedError as isize,
+        FileNotFound = kAudio_FileNotFoundError as isize,
+        FilePermission = kAudio_FilePermissionError as isize,
+        TooManyFilesOpen = kAudio_TooManyFilesOpenError as isize,
+        BadFilePath = kAudio_BadFilePathError as isize,
+        Param = kAudio_ParamError as isize,
+        MemFull = kAudio_MemFullError as isize,
         Unknown,
     }
 
     impl Error {
         pub fn from_os_status(os_status: OSStatus) -> Result<(), Error> {
             match os_status {
                 0 => Ok(()),
-                -4 => Err(Error::Unimplemented),
-                -43 => Err(Error::FileNotFound),
-                -54 => Err(Error::FilePermission),
-                -42 => Err(Error::TooManyFilesOpen),
-                561017960 => Err(Error::BadFilePath),
-                -50 => Err(Error::Param),
-                -108 => Err(Error::MemFull),
+                _ if os_status == kAudio_UnimplementedError => Err(Error::Unimplemented),
+                _ if os_status == kAudio_FileNotFoundError => Err(Error::FileNotFound),
+                _ if os_status == kAudio_FilePermissionError => Err(Error::FilePermission),
+                _ if os_status == kAudio_TooManyFilesOpenError => Err(Error::TooManyFilesOpen),
+                _ if os_status == kAudio_BadFilePathError => Err(Error::BadFilePath),
+                _ if os_status == kAudio_ParamError => Err(Error::Param),
+                _ if os_status == kAudio_MemFullError => Err(Error::MemFull),
```

Comment:
I find this gross but don't have a better suggestion.

Reply:
Yes it's really ugly. I have the same here: https://github.com/HEnquist/wasapi-rs/blob/091e0da655a9a33eaeab035f3e373fdd34d5f6f3/src/api.rs#L464
I'll play around with making a macro for this, should be fairly straightforward to make something that isn't so painful to look at.